common/persistence/historyStore.go

```diff
@@ -408,83 +402,100 @@ func (m *historyV2ManagerImpl) readRawHistoryBranch(
 func (m *historyV2ManagerImpl) readHistoryBranch(
 	byBatch bool,
 	request *ReadHistoryBranchRequest,
-) ([]*historypb.HistoryEvent, []*historypb.History, []byte, int, int64, error) {
+) ([]*historypb.HistoryEvent, []*historypb.History, []byte, int, int64, int64, error) {
```

Comment:
We really need to convert the return params to a struct.

Reply:
This is an internal method that's used just in two places here, I'd probably not bother packing/unpacking it into the struct.
Let me know if you feel strongly about this then I will do it.